File: dot.py
class Dot2d:
    def __init__(self, x=0, y=0): 
        try:
            self.__x = float(x) # Инициализация точки х
            self.__y = float(y) # Инициализация точки y
        except Exception:
            # Кидаем ошибку, если переданные данные не являются числами
            raise ValueError("Invalid values")
    # Вывод координат точки в консоль
    def printDot(self):
        print("Координаты точки по x: ", self.__x)
        print("Координаты точки по y: ", self.__y)
    # Геттер x
    def get__x(self):
        return self.__x
    # Геттер у
    def get__y(self):
        return self.__y

class Dot3d(Dot2d):
    def __init__(self, x=0, y=0, z=0):
        try:
            Dot2d.__init__(self, x, y)
            self.__z = z # Инициализация точки z
        except Exception:
            # Кидаем ошибку, если переданные данные не являются числами
            raise ValueError("Invalid values")
    # Вывод координат точки в консоль
    def printDot(self):
        print("Координаты точки по x: ", self.get__x())
        print("Координаты точки по y: ", self.get__y())
        print("Координаты точки по z: ", self.__z)

File: dot.cpp
#include<iostream>

// Класс, который описывает точку в двумерном пространстве 
class Dot2d
{
private:
    float x; // Координаты точки по х
    float y; // Координаты точки по у
public:
    Dot2d();
    Dot2d(float x, float y);
    virtual ~Dot2d();
    virtual void print(); 
    float getX() {
        return x;
    }
    float getY() {
        return y;
    }
};

Dot2d::Dot2d()
{
    this->x = 0;
    this->y = 0;
}

Dot2d::Dot2d(float x, float y) {
    // Инициализация координат точки
    this->x = x;
    this->y = y;
}

Dot2d::~Dot2d()
{
}

// Вывод точки в консоль
void Dot2d::print() {
    std::cout << " Координаты x: " << x << " Координаты y: " << y << std::endl;
}

// Класс, который описывает точку в трёхмерном пространстве
class Dot3d: public Dot2d {
    private: 
        float z; // Координаты точки по z
    public:
        Dot3d();
        Dot3d(float x, float y, float z) : Dot2d(x, y) {
            //Инициализация координат точки
            this->z = z;
        }
        ~Dot3d();
        void print();
        float getZ() {
            return z;
        }
};

Dot3d::Dot3d()
{
    this->z = 0;
}


Dot3d::~Dot3d() {

}

// Вывод координат точки в консоль
void Dot3d::print() {
    std::cout << "Координаты x: " << getX() << " Координаты y: " << getY() << " Координаты z: " << z << std::endl;
}
